examples/tensorflow/image_recognition/keras_models/densenet169/quantization/qat/main.py

```diff
@@ -0,0 +1,185 @@
+        latency_list = []
+        for idx, (inputs, labels) in enumerate(dataloader):
+            # dataloader should keep the order and len of inputs same with input_tensor
+            inputs = np.array([inputs])
+            assert len(input_tensor) == len(inputs), \
+                'inputs len must equal with input_tensor'
+            feed_dict = dict(zip(input_tensor, inputs))
+
+            start = time.time()
+            predictions = model.sess.run(output_tensor, feed_dict)
+            end = time.time()
+
+            predictions, labels = postprocess((predictions, labels))
+            metric.update(predictions, labels)
+            latency_list.append(end-start)
+            if idx + 1 == iteration:
+                break
+        latency = np.array(latency_list).mean() / FLAGS.batch_size
+        return latency
+
+    from neural_compressor.experimental.data.dataloaders.default_dataloader import DefaultDataLoader
+    dataset = TensorflowImageRecord(root=FLAGS.dataset_location, transform=ComposeTransform(transform_list=[
+            TensorflowResizeCropImagenetTransform(height=224, width=224)]))
+    dataloader = DefaultDataLoader(dataset, batch_size=FLAGS.batch_size)
+    latency = eval_func(dataloader)
+    if FLAGS.benchmark and FLAGS.mode == 'performance':
+        print("Batch size = {}".format(FLAGS.batch_size))
+        print("Latency: {:.3f} ms".format(latency * 1000))
+        print("Throughput: {:.3f} images/sec".format(1. / latency))
+    acc = metric.result()
+    return acc
+
+
+def main():
+    if FLAGS.tune:
+        logger.info('start quantizing the model...')
+        from neural_compressor import training, QuantizationAwareTrainingConfig
+        config = QuantizationAwareTrainingConfig()
+        compression_manager = training.prepare_compression(FLAGS.input_model, config)
+        compression_manager.callbacks.on_train_begin()
+
+        q_aware_model = compression_manager.model.model
+        q_aware_model.compile(
+            optimizer='sgd',
+            loss=tf.keras.losses.SparseCategoricalCrossentropy(),
+            metrics=["accuracy"],
+        )
+
+        q_aware_model.summary()
+        x_train, y_train = prepare_data(FLAGS.dataset_location)
+        q_aware_model.fit(x_train, y_train, batch_size=64, epochs=1)
+
+        compression_manager.callbacks.on_train_end()
+        compression_manager.save(FLAGS.output_model)
+
+    if FLAGS.benchmark:
+        from neural_compressor.benchmark import fit
+        from neural_compressor.experimental import common
+        from neural_compressor.config import BenchmarkConfig
+        assert FLAGS.mode == 'performance' or FLAGS.mode == 'accuracy', \
+        "Benchmark only supports performance or accuracy mode."
+
+        model = common.Model(FLAGS.input_model).graph_def
+        if FLAGS.mode == 'performance':
+            conf = BenchmarkConfig(cores_per_instance=4, num_of_instance=7)
+            fit(model, conf, b_func=evaluate)
```

Comment:
This example use b_func for benchmark, so please input '--iters' in 'run_benchmark.py' for performance mode.